### PR TITLE
ordering reversed in yaml list

### PR DESCRIPTION
**Pull request by cedrictools**

### CODE REVIEW

File: src/main/clojure/confucius/utils.clj [hunk 1]
```diff
@@ -14,6 +14,7 @@
     (seq? form) (outer (doall (map inner form)))
     (instance? clojure.lang.IRecord form) (outer (reduce (fn [r x] (conj r (inner x))) form form))
     (instance? java.util.LinkedHashMap form) (outer (into {} (map inner (into {} form))))
+    (instance? java.util.ArrayList form) (outer (into [] (map inner form)))
```
Comment (instilled) [inline src/main/clojure/confucius/utils.clj:17]: same as fixing the list below...

**Comment by cedrictools** [comment]
same indeed, since this bug is more about yaml parsing than "walk", I had a look a snakeyaml based clojure library and ArrayList is what I've found here
https://github.com/owainlewis/yaml/blob/master/src/yaml/reader.clj#L29
https://github.com/lancepantz/clj-yaml/blob/master/src/clj_yaml/core.clj#L57

**Comment by instilled** [comment]
Not sure why the clj yml parsers you mentioned don't use `java.util.List`. Using `java.util.List` would be perfectly fine and the code would work with all kinds of data structures obeying the semantics of `java.util.List`: `Arraylist`, `LinkedList`, etc.

**Comment by cedrictools** [comment]
clj-yaml and clojure.walk would probably also work fine, orthogonal features 
was too lazy for a pragmatic pull request

**Pull request closed without merging**